test/integration/smoke/test_vpc_vpn.py

```diff
@@ -570,127 +584,139 @@ def test_vpc_site2site_vpn(self):
-        self.debug("VPN customer gateway added for VPC %s enabled" % vpc2.id )
+        customer2_response = VpnCustomerGateway.create(
+            self.apiclient, services, "Peer VPC2", ip2.ipaddress, vpc2.cidr, self.account.name, self.domain.id)
+        self.debug("VPN customer gateway added for VPC %s enabled" % vpc2.id)
         self.logger.debug(vars(customer2_response))
 
         # 6) Connect two VPCs
-        vpnconn1_response = Vpn.createVpnConnection(self.apiclient, customer1_response.id, vpn2_response['id'], True)
+        vpnconn1_response = Vpn.createVpnConnection(
+            self.apiclient, customer1_response.id, vpn2_response['id'], True)
         self.debug("VPN passive connection created for VPC %s" % vpc2.id)
 
-        vpnconn2_response = Vpn.createVpnConnection(self.apiclient, customer2_response.id, vpn1_response['id'])
+        vpnconn2_response = Vpn.createVpnConnection(
+            self.apiclient, customer2_response.id, vpn1_response['id'])
         self.debug("VPN connection created for VPC %s" % vpc1.id)
 
-        self.assertEqual(vpnconn2_response['state'], "Connected", "Failed to connect between VPCs!")
+        self.assertEqual(
+            vpnconn2_response['state'], "Connected", "Failed to connect between VPCs!")
 
         # acquire an extra ip address to use to ssh into vm2
         try:
             vm2.public_ip = PublicIPAddress.create(
-                                    apiclient=self.apiclient,
-                                    accountid=self.account.name,
-                                    zoneid=self.zone.id,
-                                    domainid=self.account.domainid,
-                                    services=self.services,
-                                    networkid=ntwk2.id,
-                                    vpcid=vpc2.id)
+                apiclient=self.apiclient,
+                accountid=self.account.name,
+                zoneid=self.zone.id,
+                domainid=self.account.domainid,
+                services=self.services,
+                networkid=ntwk2.id,
+                vpcid=vpc2.id)
         except Exception as e:
             self.fail(e)
         finally:
-            self.assert_(vm2.public_ip is not None, "Failed to aqcuire public ip for vm2")
-
+            self.assert_(
+                vm2.public_ip is not None, "Failed to aqcuire public ip for vm2")
 
         # Create port forward to be able to ssh into vm2
         try:
-            natrule = self.create_natrule(vpc2, vm2, 22, 22, vm2.public_ip, ntwk2)
+            natrule = self.create_natrule(
+                vpc2, vm2, 22, 22, vm2.public_ip, ntwk2)
         except Exception as e:
             self.fail(e)
         finally:
-            self.assert_(natrule is not None, "Failed to create portforward for vm2")
-            time.sleep(10)
+            self.assert_(
+                natrule is not None, "Failed to create portforward for vm2")
+            time.sleep(20)
```

Comment:
nope not the essence of this change, that was the waiting for the template download which has been implemented using the template.download method.

the time sleep is to catch anything else that could be slow in certain environments, such as vm deployment.